overlord/restart/restart.go

```diff
@@ -299,46 +315,10 @@ func notifyRebootRequiredClassic(rebootRequiredSnap string) error {
 	return nil
 }
 
-// FinishTaskWithRestart will finish a task that needs a restart, by setting
-// its status and requesting a restart.
-// It should usually be invoked returning its result immediately from the
-// caller.
-// In some situations it might not set the desired status directly and schedule
-// the restart. If a manual restart is preferred (like on classic, where
-// automatic restarts are undesirable) it will instead set the task to
-// WaitStatus and return a marker error of type state.Wait.
-// The restart manager itself will then make sure to set the the status as
-// requested later on system restart to allow progress again.
-func FinishTaskWithRestart(task *state.Task, status state.Status, rt RestartType, snapName string, rebootInfo *boot.RebootInfo) error {
```

Comment:
why the change of name?

Reply:
Originally during I wrote this, this co-existed with `FinishTaskWithRestart`, I renamed this back to reduce the number of changes.